include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -564,6 +568,111 @@ __pattern_minmax_element(_ExecutionPolicy&& __exec, _Range&& __rng, _Compare __c
+        ::std::forward<_ExecutionPolicy>(__exec), sizeof(__key_type) + sizeof(__val_type), __wgroup_size);
+
+    // element is copied if it is the last element (marks end of final segment), is in an index
+    // evenly divisible by wg size (ensures segments are not long), or has a key not equal to the
+    // adjacenent element (marks end of real segments)
+    auto __res = __pattern_copy_if(::std::forward<_ExecutionPolicy>(__exec), __view1, __view2,
+                                   [__n, __binary_pred, __wgroup_size](const auto& __a) {
+                                       return ::std::get<0>(__a) == __n ||
+                                              ::std::get<0>(__a) % __wgroup_size == 0 ||              // segment size
+                                              !__binary_pred(::std::get<1>(__a), ::std::get<2>(__a)); //keys comparison
+                                   },
+                                   unseq_backend::__brick_assign{});
+
+    //reduce by segment
+    oneapi::dpl::__par_backend_hetero::__parallel_for(
+        ::std::forward<_ExecutionPolicy>(__exec), unseq_backend::__brick_reduce_idx<_BinaryOperator>{__binary_op},
+        __res, experimental::ranges::views::all_read(__idx), ::std::forward<_Range2>(__values),
+        experimental::ranges::views::all_write(__tmp_out_values))
+        .wait();
+
+    // Round 2: final reduction to get result for each segment of equal adjacent keys
+
+    // create views over adjacent keys
+    oneapi::dpl::__ranges::all_view<__key_type, __par_backend_hetero::access_mode::read_write> __new_keys(
+        __tmp_out_keys);
+    auto __k3 = oneapi::dpl::__ranges::drop_view_simple<
+        oneapi::dpl::__ranges::all_view<__key_type, __par_backend_hetero::access_mode::read_write>, __diff_type>(
+        __new_keys, 1);
+    auto __k4 = __new_keys;
+
+    // view3 elements are a tuple of the element index and pairs of adjacent keys
+    // view4 elements are a tuple of the elements where key-index pairs will be written by copy_if
+    auto __view3 = experimental::ranges::zip_view(experimental::ranges::views::iota(0, __res), __k3, __k4);
+    auto __view4 = experimental::ranges::zip_view(experimental::ranges::views::all_write(__out_keys),
+                                                  experimental::ranges::views::all_write(__idx));
+
+    // element is copied if it is the last element (end of final segment), or has a key not equal to
+    // the adjacenent element (end of a segment). Artificial segments based on wg size are not created.
+    using __new_name = oneapi::dpl::__par_backend_hetero::__new_kernel_name<_ExecutionPolicy, 1>;
+    auto __new_exec = oneapi::dpl::execution::make_hetero_policy<__new_name>(::std::forward<_ExecutionPolicy>(__exec));
+    __res = __pattern_copy_if(__new_exec, __view3, __view4,
+                              [__res, __binary_pred](const auto& __a) {
+                                  return ::std::get<0>(__a) == __res ||
```

Comment:
Similarly to line 626, this condition seems to be always false.

Reply:
Removed.